### DISCUSSION TITLE
[Feature request] Chunk echo, output, warnings etc to different tabs

### DISCUSSION THREAD

**Discussion by sorenoneill** [quarto-dev/quarto-cli#3971] [Feature Requests]
# Background 
I'm preparing materials for a course on the R ecosystem for PhD students .. so naturally, I prepare my presentations as quarto documents.

I'm currently using callouts with collapse=true .. and it works just fine, albeit the expanding/collapsing does make it a bit difficult for the students follow the plot, when the text moves around with expand/collapse.

# Suggested feature

I suggest a feature to split the output generated by a chunk into tabs -- by output I mean i) the text/tables/figures generated by code, ii) the echoed code it self, iii) any warnings or errors.

This would make for a neat way to present code and output in a visually pleasing way in presentations, webpages, etc

**Comment by mcanouil** [comment]
How Quarto would guess the type of content in a code cell (R, Python, Julia, Observable, ...)? 🤔

I believe this is more about how you design your slide, using "layout" options (especially in `revealjs` format`) and "tabset".

Also, you can use `classes` to add CSS class to a code cell, and you can use any of the `class-*` and `attr-*` option to add particular CSS classes or HTML attributes to some specific part, see <https://quarto.org/docs/reference/cells/cells-knitr.html>.

Another approach if to generate the content how you want it to appear using `results: asis` and `knitr::knit_expand` or `knitr::knit_child()`.

**Comment by cscheid** [comment]
This is likely possible to do today with quarto 1.3 and Lua filters. You could start by looking at the Pandoc native output with `quarto render --to native`, then write a Lua filter that detects the divs and converts them to a `quarto.Tabset`.
